analyzer/src/main/kotlin/PackageCurationProvider.kt

```diff
@@ -24,22 +24,45 @@ import org.ossreviewtoolkit.model.PackageCuration
 import org.ossreviewtoolkit.model.config.PackageCurationProviderConfiguration
 import org.ossreviewtoolkit.utils.common.ConfigurablePluginFactory
 import org.ossreviewtoolkit.utils.common.Plugin
+import org.ossreviewtoolkit.utils.common.getDuplicates
 
 /**
  * The extension point for [PackageCurationProvider]s.
  */
 interface PackageCurationProviderFactory<CONFIG> : ConfigurablePluginFactory<PackageCurationProvider> {
     companion object {
         val ALL = Plugin.getAll<PackageCurationProviderFactory<*>>()
+        const val ORT_YML_PROVIDER_ID = "ort-yml"
 
         /**
-         * Return a new provider instance for each [enabled][PackageCurationProviderConfiguration.enabled] provider
-         * configuration in [configurations]. The given [configurations] must be ordered highest-priority first while
-         * the returned providers are ordered lowest-priority first, which is the order in which the corresponding
-         * curations must be applied.
+         * Return a new (identifier, provider instance) tuple for each
+         * [enabled][PackageCurationProviderConfiguration.enabled] provider configuration in [configurations]. The given
+         * [configurations] must be ordered highest-priority first while the returned providers are ordered
+         * lowest-priority first, which is the order in which the corresponding curations must be applied.
          */
-        fun create(configurations: List<PackageCurationProviderConfiguration>): List<PackageCurationProvider> =
-            configurations.filter { it.enabled }.map { ALL.getValue(it.type).create(it.config) }.asReversed()
+        fun create(
+            configurations: List<PackageCurationProviderConfiguration>
+        ): List<Pair<String, PackageCurationProvider>> =
```

Comment:
I'm surprised that this now is a list of pairs instead of a map [as discussed before](https://github.com/oss-review-toolkit/ort/pull/6375/files#diff-f97d611d1c2477b31e641e58a37794afd7b77e3e586b8cc2660562c56f85ebddR40). Have you considered it to be a map?

Reply:
I was reminded that, as order of curation providers matters, a general map (with unordered keys) is not suitable.